js/main.js

```diff
@@ -41,3 +50,48 @@ function handleSearch(entry){
         entry.classList.add('hidden')
       }
     }
+
+const filterIcon=document.getElementById('filter-icon')
+const typeFilterContainer=document.getElementById('type-filter-container')
+filterIcon.addEventListener('click',()=>{
+  if(typeFilterContainer.style.display==='flex'){
+    typeFilterContainer.style.display='none'
+  }
+  else{
+    typeFilterContainer.style.display='flex'
+  }
+
+})
+
+const typeCheckboxes=document.querySelectorAll('.checkbox')
+function handleTypeFilter(event){
+  if(event.target.className==='checkbox'){
+    let counter=0
+    typeCheckboxes.forEach((value)=>{
+      if(value.checked===true){
+        listItems.forEach((element)=>{
+        element.classList.add('hidden')
+            })
+        getPokemonByType(value.name)}
+      else{
+        counter++
+        if (counter===18){
+          listItems.forEach((element)=>{
+          element.classList.remove('hidden')
```

Comment:
This counter logic seems more complicated than it needs to be. 

I would create a function `clearTypeFilters` instead